analysis.tex

```diff
@@ -1,29 +1,53 @@
 \section{Analysis}
 
-\begin{theorem}
-  If $H$ is modelled as a \emph{Random Oracle}, then the protocol $\Pi$ of Section~\ref{section:construction} is \emph{unspendable}.
+\begin{theorem}[Correctness]
+  The Proof-of-Burn protocol $\Pi$ of Section~\ref{section:construction} is \emph{correct}.
 \end{theorem}
 \begin{proof}
-  $H(t) \xor 1 = pkh$.
-  Adversary knows a $pk$ st $H(pk) = pkh$.
-  This is impossible under the Random Oracle model.
-  TODO
+  We need to prove that $\forall \kappa, t. \BurnVerify(1^\kappa, t, \GenBurnAddress(1^\kappa, t)) = \textsf{true}$.
+
+  Based on Algorithm~\ref{alg.construction}, $\BurnVerify(1^\kappa, t, \GenBurnAddress(1^\kappa, t)) = \textsf{true}$ if and only if $\GenBurnAddress(1^\kappa, t) = \GenBurnAddress(1^\kappa, t)$, which always holds as $\GenBurnAddress$ is deterministic.
+\end{proof}
+
+\begin{theorem}[Unspendability]
+  If $H$ is modelled as a preimage-resistant hash function, then the protocol $\Pi$ of Section~\ref{section:construction} is \emph{unspendable}.
+\end{theorem}
+\begin{proof}
+  Assume that the protocol $\Pi$ is not unspendable. This means that there exists a probabilistic polynomial-time adversary $\mathcal{A}$ such that:
+
+  \[
+    \Pr[\textsc{spend-attack}_{\mathcal{A},
+    \Pi}(\kappa) = \textsf{true}] > negl(\kappa)
+  \]
+
+  This would imply that the adversary can generate $t$, $m$, $\sigma = (pk, \sigma')$ and $pkh$ such that $\BurnVerify(1^\kappa, t, pkh) = \true \land \SpendVerify(m, \sigma, pkh) = \true$ with non-negligible probability in $\kappa$.
```

Comment:
I think at this point it is useful to condition on the event that A is successful.